File: tba/src/main/java/de/gaalop/productcomputer2/Algebra5d.java
package de.gaalop.productcomputer2;

import java.util.HashMap;
import java.util.LinkedList;

/**
 *
 * @author christian
 */
public class Algebra5d extends Algebra {

    @Override
    public void create() {
        base = new String[]{"e1","e2","e3","einf","e0"};
        base2 = new String[]{"e1","e2","e3","ep","em"};
        baseSquaresStr = new HashMap<String, Integer>();
        baseSquaresStr.put("e1", 1);
        baseSquaresStr.put("e2", 1);
        baseSquaresStr.put("e3", 1);
        baseSquaresStr.put("ep", 1);
        baseSquaresStr.put("em", -1);

        mapToPlusMinus = new HashMap<String, LinkedList<BladeStr>>();

        //e0(em,ep)
        LinkedList<BladeStr> listE0 = new LinkedList<BladeStr>();
        listE0.add(new BladeStr(0.5f, "em"));
        listE0.add(new BladeStr(-0.5f, "ep"));
        mapToPlusMinus.put("e0", listE0);
        //einf(em,ep)
        LinkedList<BladeStr> listEinf = new LinkedList<BladeStr>();
        listEinf.add(new BladeStr("em"));
        listEinf.add(new BladeStr("ep"));
        mapToPlusMinus.put("einf", listEinf);

        mapToZeroInf = new HashMap<String, LinkedList<BladeStr>>();

        //em(einf,e0)
        LinkedList<BladeStr> listEm = new LinkedList<BladeStr>();
        listEm.add(new BladeStr(0.5f, "einf"));
        listEm.add(new BladeStr("e0"));
        mapToZeroInf.put("em", listEm);
        //ep(einf,e0)
        LinkedList<BladeStr> listEp = new LinkedList<BladeStr>();
        listEp.add(new BladeStr(0.5f, "einf"));
        listEp.add(new BladeStr(-1, "e0"));
        mapToZeroInf.put("ep", listEp);

    }

}

File: productComputerC/Algebra5d.cpp
/*
 * Algebra5d.cpp
 *
 *  Created on: 22.11.2011
 *      Author: christian
 */

#include "Algebra5d.h"

Algebra5d::Algebra5d() {
}

Algebra5d::~Algebra5d() {
}

void Algebra5d::create() {
	base.push_back("e1");
	base.push_back("e2");
	base.push_back("e3");
	base.push_back("einf");
	base.push_back("e0");

	base2.push_back("e1");
	base2.push_back("e2");
	base2.push_back("e3");
	base2.push_back("ep");
	base2.push_back("em");

	baseSquaresStr["e1"] = 1;
	baseSquaresStr["e2"] = 1;
	baseSquaresStr["e3"] = 1;
	baseSquaresStr["ep"] = 1;
	baseSquaresStr["em"] = -1;

	//e0(em,ep)
	ListOfBladeStr listE0;
	string em = "em";
	string ep = "ep";
	string e0 = "e0";
	string einf = "einf";
	listE0.push_back(BladeStr(0.5,em));
	listE0.push_back(BladeStr(-0.5,ep));
	mapToPlusMinus[e0] = listE0;

	//einf(em,ep)
	ListOfBladeStr listEinf;
	listEinf.push_back(BladeStr(em));
	listEinf.push_back(BladeStr(ep));
	mapToPlusMinus[einf] = listEinf;


	//em(einf,e0)
	ListOfBladeStr listEm;
	listEm.push_back(BladeStr(0.5f, einf));
	listEm.push_back(BladeStr(e0));
	mapToZeroInf[em] = listEm;

	//ep(einf,e0)
	ListOfBladeStr listEp;
	listEp.push_back(BladeStr(0.5f, einf));
	listEp.push_back(BladeStr(-1, e0));
	mapToZeroInf[ep] = listEp;

}
